packages/core-data/src/reducer.js

```diff
@@ -277,6 +369,82 @@ export function autosaves( state = {}, action ) {
 	return state;
 }
 
+export const undo = combineReducers( {
```

Comment:
I was doubtful about the refactoring to rely on actions instead of history for this but just the fact that it makes the implementation "separate" from the original reducer (SoC) seems cleaner.